examples/41_fused_multi_head_attention/gemm_kernel_utils.h

```diff
@@ -141,6 +141,37 @@ constexpr CUTLASS_HOST_DEVICE integer align_up(integer n, integer m) {
   return ((n + m - 1) / m) * m;
 }
 
+inline int32_t getMaximumSharedMemoryPerBlockKb(int cc) {
```

Comment:
@jackkosaian , there is a cuda api call that can get these numbers, correct?

Reply:
i think we can query device property and get `sharedMemPerBlockOptin`

Reply:
@danthe3rd , if we can get the value from a cuda call, is it better to call the api instead of looking up a table here?

Reply:
one example usage is https://github.com/NVIDIA/cutlass/blob/master/test/unit/conv/device/conv2d_testbed.h#L211

Reply:
Yes, `cudaGetDeviceProperties(properties, ...)` and the `properties.sharedMemPerBlockOptin` should be what is needed.

Reply:
@danthe3rd, is `getMaximumSharedMemoryPerBlockKb()` expected to be called before every kernel launch? If so, and if changing to use the CUDA API, it may be beneficial to restructure the code so that `cudaGetDeviceProperties()` is called only once and the `properties.sharedMemPerBlockOptin` property is saved somewhere. This avoids having the API call on the critical path.

Reply:
You might want to use `cudaDeviceGetAttribute` for better performance. See [blog ](https://developer.nvidia.com/blog/cuda-pro-tip-the-fast-way-to-query-device-properties/) for more details